hilti/toolchain/src/ast/meta.cc

```diff
@@ -5,3 +5,46 @@
 using namespace hilti;
 
 std::unordered_set<Meta> Meta::_cache;
+
+Meta Meta::combineLocs(const Meta* other) const {
+    if ( location().file() != other->location().file() || location() == other->location() )
+        return *this;
+
+    // We have to be careful about column numbers when the line changes
+    int from_line = -1;
+    int to_line = -1;
+    int from_character = -1;
+    int to_character = -1;
+
+    // These calculations are weird. The logic:
+    //
+    // 1) If this starts on a line before other, then choose this for the first point
+    // 2) If this and other are same line, choose this if it is a lower column
+    // 3) Otherwise, choose other
+    //
+    // Then the same logic applies to 'to' as well
```

Comment:
Please check this logic, this is a gosh darn mess but I dunno how to do it better. :/